[DOCS-8074] Add Obs Pipelines Monitoring doc (#25263)
DataDog/documentation: content/en/observability_pipelines/monitoring/metrics.md

---
title: Metrics
disable_toc: false
aliases:

> - /path-to-old-doc/

Reviewer: this needs to be updated or the section removed :)

Author: Removed

further_reading:
- link: "/metrics/summary/"
  tag: "Documentation"
  text: "Learn more about the Metrics Summary"
- link: "/metrics/explorer/"
  tag: "Documentation"
  text: "Using the Metrics Explorer to explore and analyze your metrics"
- link: "/getting_started/dashboards/"
  tag: "Documentation"
  text: "Getting started with dashboards"
- link: "/getting_started/monitors/"
  tag: "Documentation"
  text: "Getting started with monitors"
---

## Overview

> Use Observability Pipelines' metrics to create [dashboards][1], [notebooks][2], and [monitors][3] for you specific use case. Use [Metrics Summary][5] to see metadata and tags available for the metrics. You can also see which dashboards, notebooks, monitors, and SLOs are using those metrics. See [Getting Started with Tags][4] for more information on how to use tags to group metrics by specific pipelines, Workers, and components.

Reviewer: ```suggestion
Use metrics from Observability Pipelines to create [dashboards][1], [notebooks][2], and [monitors][3] for your specific use case. Use [Metrics Summary][5] to see metadata and tags available for the metrics. You can also see which dashboards, notebooks, monitors, and SLOs are using those metrics. See [Getting Started with Tags][4] for more information on how to use tags to group metrics by specific pipelines, Workers, and components.
```

> Use Observability Pipelines' metrics to create [dashboards][1], [notebooks][2], and [monitors][3] for you specific use case. Use [Metrics Summary][5] to see metadata and tags available for the metrics. You can also see which dashboards, notebooks, monitors, and SLOs are using those metrics. See [Getting Started with Tags][4] for more information on how to use tags to group metrics by specific pipelines, Workers, and components.

Reviewer: Might this make sense as a bulleted list? If not, or even if so, I think this needs an introductory sentence about this page itself.

Author: I added a sentence and make it in a bulleted list.

## Pipeline metrics

Bytes in per second
: **Metrics**: `pipelines.host.network_receive_bytes_total`
: **Description:** The number of events the pipeline receives per second.

Bytes out per second
: **Metrics**: `pipelines.host.network_receive_bytes_total`
: **Description:** The number of bytes the pipeline receives per second.

## Component metrics

These metrics are available for sources, processors, and destinations.

> **Note**: `pipelines.utilization` is only available for processors and destinations.

Reviewer: This is repetitive with the information below. I feel like you could just remove it.

Author: Removed it.

Bytes in per second
: **Metric**: `pipelines.component_received_bytes_total`
: **Description**: The number of bytes the component receives per second.
: **Available for**: Sources, processors, and destinations.

Bytes out per second
: **Metric**: `pipelines.component_sent_event_bytes_total`
: **Description**: The number of bytes the component sends to the destinations.
: **Available for**: Sources, processors, and destinations.

Events in per second
: **Metric**: `pipelines.component_received_event_bytes_total`
: **Description**: The number of events the component receives per second.
: **Available for**: Sources, processors, and destinations.

Events out per second
: **Metric**: `pipelines.component_sent_event_bytes_total`
: **Description**: The number of events the component sends to the destinations.
: **Available for**: Sources, processors, and destinations.

Errors
: **Metric**: `pipelines.component_errors_total`
: **Description**: The number of errors encountered by the component.
: **Available for**: Sources, processors, and destinations.

Data dropped intentionally or unintentionally
: **Metric**: `pipelines.component_discarded_events_total`

> : **Description**: The number of events dropped. Break down the metric by using the `intentional:true` tag to filter for events that are intentionally dropped or the `intentional:false` tag for events that are not intentionally dropped.

Reviewer: ```suggestion
: **Description**: The number of events dropped. **Note**: To break down this metric, use the `intentional:true` tag to filter for events that are intentionally dropped or the `intentional:false` tag for events that are not intentionally dropped.
```

: **Available for**: Sources, processors, and destinations.

Utilization
: **Metric**: `pipelines.utilization`

> : **Description**: The component's activity. A value of `0` indicates an idle component that is waiting for input. A value of `1` indicates a component that is never idle. A value of `1` indicates that the component is busy and likely a bottleneck in the processing topology and creating backpressure, which could cause events to be dropped.

Reviewer: ```suggestion
: **Description**: The component's activity. A value of `0` indicates an idle component that is waiting for input. A value of `1` indicates a component that is never idle, which means that the component is likely a bottleneck in the processing topology that is creating backpressure, which might cause events to be dropped.
```
